articles/2016-09-01-semantic-html-tags.md

```diff
@@ -0,0 +1,98 @@
+
+Screen readers, parsers, and search engines that try to read your HTML and attempt to understand its structure. If you use semantic tags, you are giving these systems an easier time. Your audience is diverse and has a wide range of abilities/needs that you must address by including [ARIA attributes](http://html5doctor.com/using-aria-in-html/) in your HTML code. These attributes are necessary to ensure that accessibility API's can get the relevant information about your page. As of HTML5, semantic elements have the ARIA *role* built in by default. While it is important to consider ARIA attributes and include them whenever possible, using the right element for the content is a good bare minimum.
+
+#### Readability
+
+It also makes your code significantly easier to read by other humans. In a complex or large page with levels and levels of `<div>`s, it becomes impossible to understand the flow of information. Too many `<div>`s can fill up a page until it becomes unreadable. This is commonly referred to as _divitis_. Avoid it at all costs.
+
+However, it is much easier to understand if you take advantage of elements like `<main>`, `<section>`, `<header>`, etc. because it helps you visualize the flow of information as you read through the HTML.
+
+#### Styling
+
+An additional benefit of using semantic tags, is that you can write more general CSS selectors. For a refresher on specificity of CSS selectors, read the [blog post](2016-08-02-why-you-should-never-use-important.md) I wrote on the subject). The benefit of general selectors is that you can establish a foundation of styles (for example, all `<section>`s have a blue background and 10 pixels of padding), that can easily be overwritten in a more specific situation, without having to add too many extra classes or incorporating `!important`.
+
+It should be stated that this approach only works in certain cases, usually smaller or simpler UI's. Styling on tag selectors is considered a bad practice if the UI is too large to be appropriately represented as such. There are situations where it can work, but you are more likely going to need to use class selectors for your styling.  
+
+### Frameworks and Templating
+
+If you're using a front-end framework like Ember, you will notice that a lot of components will be wrapped in `<div>`s. This is ok, as the benefit of a `<div>` is that it's generic enough to be used everywhere. If you are committed to using semantic tags, you have two choices here. You can either wrap the template inside the component with the appropriate tag,
+
+```handlebars
+{{#some-component}}
+  <section>
+    <!-- section content goes here -->
+  </section>
+{{/some-component}}
+```
+
+This will render to the following HTML
+
+```html
+<div id="ember123">
+  <section>
+    <!-- section content goes here -->
+  </section>
+</div>
+```
+
+Another option (one that I personally prefer), is to change the `tagName` attribute of the component to match the appropriate semantic tag,
```

Comment:
I think you should make a stronger statement here, this seems like a far better solution.